bam/reader.go

```diff
@@ -109,90 +160,63 @@ func (br *Reader) Read() (*sam.Record, error) {
 		return nil, io.EOF
 	}
 
-	r := errReader{r: br.r}
-	bin := binaryReader{r: &r}
-
-	// Read record header data.
-	blockSize := int(bin.readInt32())
-	r.n = 0 // The blocksize field is not included in the blocksize.
-
+	var buf [4]byte
+	n, err := br.r.Read(buf[:4])
```

Comment:
We need to check that this always works (the io.Reader interface doesn't guarantee it, but I'm pretty sure bgzf.Reader behaves correctly). This can happen latter, and can just be a replacement of this with a call to io.ReadFull.